trieste/acquisition/rule.py

```diff
@@ -1484,122 +1484,113 @@ def filter_datasets(
         return filtered_datasets
 
 
-class UpdatableTrustRegionBox(Box, UpdatableTrustRegion):
-    """
-    A simple updatable box search space with a centre location and an associated global search
-    space.
-    """
+class UpdatableTrustRegionWithLocationInGlobalSearchSpace(UpdatableTrustRegion):
+    """An updatable trust region with a concept of a location within a global search space."""
 
-    def __init__(
+    def _init_location(
         self,
-        global_search_space: SearchSpace,
-        region_index: Optional[int] = None,
-        input_active_dims: Optional[Union[slice, Sequence[int]]] = None,
-    ):
-        """
-        :param global_search_space: The global search space this search space lives in.
-        :param region_index: The index of the region in a multi-region search space. This is used to
-            identify the local models and datasets to use for acquisition. If `None`, the
-            global models and datasets are used.
-        :param input_active_dims: The active dimensions of the input space, either a slice or list
-            of indices into the columns of the space. If `None`, all dimensions are active.
-        """
-        Box.__init__(self, global_search_space.lower, global_search_space.upper)
-        UpdatableTrustRegion.__init__(self, region_index, input_active_dims)
-        self._global_search_space = global_search_space
-        # Random initial location in the global search space.
-        self.location = tf.squeeze(global_search_space.sample(1), axis=0)
+        models: Optional[Mapping[Tag, ProbabilisticModelType]] = None,
+        datasets: Optional[Mapping[Tag, Dataset]] = None,
+        location_candidate: Optional[TensorType] = None,
+    ) -> None:
+        # Initialize the location of the region.
```

Comment:
Given that this is now called in subclasses, it should probably have a docstring rather than a comment.